mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -249,16 +255,22 @@ object NaiveBayesModel extends Loader[NaiveBayesModel] {
       sc.parallelize(Seq(metadata), 1).saveAsTextFile(metadataPath(path))
 
       // Create Parquet data.
-      val dataRDD: DataFrame = sc.parallelize(Seq(data), 1).toDF()
-      dataRDD.write.parquet(dataPath(path))
+      val dataRDD = sc.parallelize(Seq(Row(data.labels, data.pi, data.theta)), 1)
```

Comment:
We can remove the `Data` case class and pass `labels`, `pi`, etc in directly now